### PR TITLE
[CARBONDATA-3278] Remove duplicate code to get filter string of date/timestamp

### PR DESCRIPTION
**Pull request by kevinjmh**
Remove duplicate code to get filter string of date/timestamp by method `ExpressionResult.getString()` instead

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2478/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2706/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10736/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2484/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2713/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10742/

**Comment by kevinjmh** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2720/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2489/

File: datamap/bloom/src/main/java/org/apache/carbondata/datamap/bloom/BloomCoarseGrainDataMap.java [hunk 1]
```diff
@@ -358,11 +326,12 @@ private BloomQueryModel buildQueryModelForIn(ColumnExpression ce, ListExpression
 
   private byte[] getInternalFilterValue(CarbonColumn carbonColumn, LiteralExpression le) throws
       DictionaryGenerationException, UnsupportedEncodingException {
-    Object filterLiteralValue = getLiteralExpValue(le);
     // convert the filter value to string and apply converters on it to get carbon internal value
     String strFilterValue = null;
-    if (null != filterLiteralValue) {
-      strFilterValue = String.valueOf(filterLiteralValue);
+    try {
+      strFilterValue = le.getExpressionResult().getString();
```
Comment (dhatchayani) [inline datamap/bloom/src/main/java/org/apache/carbondata/datamap/bloom/BloomCoarseGrainDataMap.java]: @kevinjmh can you please check for DATE and TIMESTAMP data type. As we expect DATE and TIMESTAMP value in Long type as filterValue and then we make StringValue out of that.
le.getExpressionResult().getString() ==> what string value this return for DATE and TIMESTAMP value?

Comment (kevinjmh) [reply datamap/bloom/src/main/java/org/apache/carbondata/datamap/bloom/BloomCoarseGrainDataMap.java]: The motivation for this PR is exactly for DATE and TIMESTAMP. You can check detail implementaion in that method

Comment (dhatchayani) [reply datamap/bloom/src/main/java/org/apache/carbondata/datamap/bloom/BloomCoarseGrainDataMap.java]: ok.

**Comment by xuchuanyin** [comment]
@kevinjmh Does this change have any impact on the legacy store?

**Comment by kevinjmh** [comment]
@xuchuanyin no, this changes only call existing codes with rich type check for literal value, the whole process is the same

**Comment by kevinjmh** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2596/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10855/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2826/

**Comment by xuchuanyin** [comment]
LGTM

**Pull request closed without merging**